packages/crawl/build.sh

```diff
@@ -0,0 +1,26 @@
+TERMUX_PKG_HOMEPAGE=https://crawl.develz.org/
+TERMUX_PKG_DESCRIPTION="Roguelike adventure through dungeons filled with dangerous monsters"
+TERMUX_PKG_LICENSE="GPL-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_SRCURL=https://github.com/crawl/crawl.git
+TERMUX_PKG_VERSION=0.29-a0
+TERMUX_PKG_GIT_BRANCH=0.29-a0
+TERMUX_PKG_DEPENDS="ncurses, zlib"
+TERMUX_PKG_BUILD_IN_SRC=true
+
+termux_step_pre_configure() {
+    export CROSSHOST="$TERMUX_HOST_PLATFORM"
+    export TERMUX_PKG_BUILDDIR="$TERMUX_PKG_SRCDIR/crawl-ref/source"
+
+    export INCLUDES_L="-I/data/data/com.termux/files/usr/include"
+    export LIBS="-llog -Wl,--rpath=/data/data/com.termux/files/usr/lib"
+}
+
+termux_step_post_configure() {
+    sed -i 's,#ifdef __ANDROID__,#ifdef __NO_THANKS__,g' "$TERMUX_PKG_BUILDDIR/syscalls.cc"
+    sed -i 's,CC = $(GCC),CC := $(CC),' "$TERMUX_PKG_BUILDDIR/Makefile"
+    sed -i 's,CXX = $(GXX),CXX := $(CXX),' "$TERMUX_PKG_BUILDDIR/Makefile"
+    sed -i '/^G.. := $(CROSSHOST)-g../d' "$TERMUX_PKG_BUILDDIR/Makefile"
+    sed -i 's,^\([^ ]\+\) := $(CROSSHOST)-\([^ ]\+\),\1 := llvm-\2,' "$TERMUX_PKG_BUILDDIR/Makefile"
+    sed -i 's,-L$(NC_PREFIX)/lib -l$(NCURSESLIB),-L/data/data/com.termux/files/usr/lib -lncurses,' "$TERMUX_PKG_BUILDDIR/Makefile"
```

Comment:
Please convert the Makefile changes to a .patch file, we can then avoid to hardcode `/data/data/com.termux/files/usr` and instead use `@TERMUX_PREFIX@` which is automatically sed'ed: https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_patch_package.sh#L27

Reply:
Ooh, thanks! That's handy~ (How do the changes I pushed look?)